src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.SpeculativeMemberSemanticModel.cs

```diff
@@ -51,6 +51,11 @@ protected override void AnalyzeBoundNodeNullability(BoundNode boundRoot, Binder
                 NullableWalker.AnalyzeWithoutRewrite(Compilation, MemberSymbol as MethodSymbol, boundRoot, binder, diagnostics, createSnapshots);
             }
 
+            protected override bool IsNullableAnalysisEnabled()
+            {
+                return Compilation.IsNullableAnalysisEnabled;
```

Comment:
>Compilation.IsNullableAnalysisEnabled; [](start = 23, length = 38)

It feels like we should use the context at the speculation position instead.

Reply:
Agreed, this should definitely be based on the current nullable state of the underlying location.

---
In reply to: [544680016](https://github.com/dotnet/roslyn/pull/49953#discussion_r544680016) [](ancestors = 544680016,544679218)

Reply:
It doesn't look like there was any response to

> And perhaps account for explicit nullable directives in the syntax we are speculating on.

---
In reply to: [546853488](https://github.com/dotnet/roslyn/pull/49953#discussion_r546853488) [](ancestors = 546853488,544680016,544679218)

Reply:
> And perhaps account for explicit nullable directives in the syntax we are speculating on.

That seems like a scenario we could support later if necessary. It would require handling `#nullable restore` as well:
```C#
#nullable enable
static void Main()
{
    /*speculative text here*/
}
```
```C#
#nullable restore
string
```

---
In reply to: [550256169](https://github.com/dotnet/roslyn/pull/49953#discussion_r550256169) [](ancestors = 550256169,546853488,544680016,544679218)

Reply:
Handled for `#nullable enable` and `disable` but not for `restore`.

---
In reply to: [550256169](https://github.com/dotnet/roslyn/pull/49953#discussion_r550256169) [](ancestors = 550256169,546853488,544680016,544679218)